src/navigations/HomeBottomTabs.js

```diff
@@ -207,135 +216,116 @@ function HomeBottomTabs({navigation}) {
+          tabBarActiveTintColor:
+            route.name === 'AnonymousChannelList' ? colors.anon_primary : colors.darkBlue
         })}>
         <Tab.Screen
-          name="Feed"
-          component={FeedScreen}
-          initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Feed')
-            // unmountOnBlur: true
-          }}
-        />
-        {/* <Tab.Screen
-          name="Feed"
-          component={WebsocketResearchScreen}
-          initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Feed')
-            // unmountOnBlur: true
-          }}
-        /> */}
-        {/* <Tab.Screen
-          name="ChannelList"
+          name="SignedChannelList"
           component={ChannelListScreen}
           initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('ChannelList'),
-            tabBarBadge:
-              unReadMessage.total_unread_count + unReadMessage.unread_post > 0
-                ? unReadMessage.total_unread_count + unReadMessage.unread_post
-                : null
-          }}
-        /> */}
+          listeners={({route}) => saveLastMenu(route)}
+          options={{tabBarBadge: signedChannelUnreadCount > 0 ? signedChannelUnreadCount : null}}
+        />
         <Tab.Screen
-          name="ChannelList"
+          name="AnonymousChannelList"
           component={ChannelListScreenV2}
           initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
           options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('ChannelList'),
-            tabBarBadge: totalUnreadCount > 0 ? totalUnreadCount : null
+            tabBarBadge: anonymousChannelUnreadCount > 0 ? anonymousChannelUnreadCount : null
           }}
         />
+        <Tab.Screen
+          name="Feed"
+          component={FeedScreen}
+          initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
+        />
         <Tab.Screen
           name="News"
           component={NewsScreen}
           initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('News')
-            // unmountOnBlur: true
-          }}
+          listeners={({route}) => saveLastMenu(route)}
         />
         <Tab.Screen
           name="Profile"
-          initialParams={{isBottomTab: true}}
           component={ProfileScreen}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Profile')
-            // unmountOnBlur:true
-          }}
+          initialParams={{isBottomTab: true}}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderTabBarIcon` function has been introduced to render different icons based on the route name. This is a good approach to make the code more modular and easier to maintain. However, ensure that all the icon components like `SignedChat`, `AnonymousChatFill`, `AnonymousChatOutline`, `MemoFeed`, `MemoNews`, and `MemoProfileIcon` are properly imported and they accept the correct props (`fill` and `stroke`). Also, make sure that the color values passed to these components are valid and defined in your `colors` utility module.



<!-- This is an auto-generated comment by OSS CodeRabbit -->